contract/src/main/kotlin/io/provenance/scope/loan/utility/ServicingDataFunctions.kt

```diff
@@ -54,7 +54,7 @@ internal fun ContractEnforcementContext.appendLoanStates(
                 existingStateTimes[effectiveTime.seconds to effectiveTime.nanos] = true
             }
         }
-    } else {
+    } else if (expectLoanStates) {
```

Comment:
> calling methods can do additional validation

This change was a result of continuing to build accomodations into helper methods I had defined for frequently reused logic — there's enough reuse of the code that I feel it's okay to add parameters like these for the sake of DRY. In this case, it's needed because
- The `AppendLoanStates` contract, which writes to the `servicingData` record, is invoked with a list of loan states which can be of any size, but we want to throw an exception if said list is empty since we'd consider it unideal to modify the record in any manner in that case
- The other contract functions which modify `servicingData` accept a `ServicingData` instance as their input, meaning there could be data other than loan states which we would want to be written to the record regardless of if loan states are provided to write as well
> Ideally this method would move on if `newLoanStates` was empty

As it currently is, it'll just initialize a few empty maps and then immediately return in that case. I'm hesitant to introduce early returns due to the intention of `validateRequirements` — to execute the cody in its body and collect all resulting violations. Introducing early returns makes it arguable harder to maintain if additional violations are added to the body.

Reply:
`to execute the cody` pls no :(